src/ui/units/dash/containers/Dialogs/DialogRelations/hooks/helpersDatasets.ts

```diff
@@ -0,0 +1,98 @@
+import {
+    DashkitMetaDataItemBase,
+    DatasetsData,
+    DatasetsFieldsListData,
+} from 'ui/components/DashKit/plugins/types';
+
+import {DashkitMetaDataItemNoRelations, Datasets, DatasetsFlatListData} from '../types';
+
+import {isControl} from './helpersControls';
+
+const mapFieldsListGuid = (list: Array<DatasetsFieldsListData>, result: Record<string, string>) => {
+    list.forEach((listItem) => {
+        if (!result[listItem.guid]) {
+            result[listItem.guid] = listItem.title || '';
+        }
+    });
+};
+
+const getDatasetItemFlatFields = (data: DatasetsData): Record<string, string> => {
+    if (!data) {
+        return {};
+    }
+    let result = {};
+    if (Array.isArray(data.fields)) {
+        mapFieldsListGuid(data.fields, result);
+    } else {
+        result = {...data.fields};
+    }
+    if (data.fieldsList) {
+        mapFieldsListGuid(data.fieldsList, result);
+    }
+    return result;
+};
+
+const getDatasetsFlatItems = (
+    datasets: DashkitMetaDataItemBase['datasets'],
+): Record<string, string> => {
+    if (!datasets) {
+        return {};
+    }
+    let result = {};
+    datasets.forEach((item) => {
+        const fields = getDatasetItemFlatFields(item);
+        result = {...result, ...fields};
+    });
+    return result;
+};
+
+const getDatasetsListWithFlatFields = (datasets: Datasets) => {
+    if (!datasets) {
+        return {};
+    }
+    const flatDatasetFields: DatasetsFlatListData = {};
+    for (const [id, data] of Object.entries(datasets)) {
+        flatDatasetFields[id] = data.fields.reduce(
+            (result, {guid, title}) => ({...result, [guid]: title}),
+            {},
+        );
```

Comment:
If you don't like desctucturing here, than it would be better change to 
```
flatDatasetFields[id] = data.fields.reduce(
    (result, {guid, title}) => { result[guid]=title; return result },
    {},
);
```
It would be equal to your's